swarm.go

```diff
@@ -283,10 +284,12 @@ func NewSwarm(config *api.Config, mockStore *mock.NodeStore) (self *Swarm, err e
 
 	if self.config.GlobalPinner {
 		lstore.WithDeliverCallback(self.pss.Deliver)
-		recoveryFunc := func(trojan.Message) {
-			// TODO: add missing chunk re-upload
+		// repairFunc takes care of re-uploading a globally pinned chunk to the network
+		repairFunc := func(m trojan.Message) {
+			chAddr := m.Payload
```

Comment:
ok but TODO: move this anonymous function into the prod package

Reply:
added